### PR TITLE
Rz aoi assist

### PR DESCRIPTION
**Pull request by rzinke**
Restructured ariaAOIassist program to exclude dates based on user input or semi-automated frame extent detection. This capability will allow the user to more efficiently create AOIs in Google Earth.

The programs now works with pandas dataframe class.

### CODE REVIEW

**Comment by dbekaert** [comment]
@rzinke can you remove the thirdparty addition of files from the PR?

**Comment by rzinke** [comment]
Hi @dbekaert, I reset my local branch and the new pull request includes changes only to the intended file. I'll keep an eye on the "Files changed" in the future.

**Comment by EJFielding** [comment]
That was a lot of notifications this morning ;-).

File: tools/bin/ariaAOIassist.py [hunk 1]
```diff
@@ -1,371 +1,505 @@
-#!/usr/bin/env python3
+#!/anaconda3/envs/ARIA-tools/bin/python3
```
Comment (dbekaert) [inline tools/bin/ariaAOIassist.py]: this is local to your machine

Comment (rzinke) [reply tools/bin/ariaAOIassist.py]: Right I noticed that, but it was written /anaconda3/... in the original code, so I must have used an older version of it. Good fix, thanks.

Comment (dbekaert) [reply tools/bin/ariaAOIassist.py]: can you correct it to what it used to be?

Comment (rzinke) [reply tools/bin/ariaAOIassist.py]: Yes, but I look at the way ariaAOIassist is currently written is the standard way, i.e., !/usr/bin/env python3. I should keep that, right?

Comment (dbekaert) [reply tools/bin/ariaAOIassist.py]: yes that woudl be consistent with the other programs:
```
#!/usr/bin/env python3
```

**Comment by dbekaert** [comment]
@rzinke would you be willing to make a small jupyter notebook to demo the functionality.
Talking a bit about the caveats on the bbox impacting the automatic generation of the AOI?

**Comment by rzinke** [comment]
Absolutely. 
I'd also be interested if anyone has better solutions for trimming the AOI.
I'll send you what I have this afternoon.

**Review by dbekaert — APPROVED** [review]
@rzinke thanks for the contribution.

I have a few potential improvements that we can purse. 
e.g. I believe the shorting of the AOI with 20km, we could do an erosion operation of 20km.
This would also impact the width of the AOI, but on the other hand i do not believe that would be a cause for problems in the processing.

**Comment by rzinke** [comment]
> @rzinke thanks for the contribution.
> 
> I have a few potential improvements that we can purse.
> e.g. I believe the shorting of the AOI with 20km, we could do an erosion operation of 20km.
> This would also impact the width of the AOI, but on the other hand i do not believe that would be a cause for problems in the processing.

This would likely be a more reliable approach than what I use currently. Do you have any ideas for how to erode the polygon?

**Comment by rzinke** [comment]
> > @rzinke thanks for the contribution.
> > I have a few potential improvements that we can purse.
> > e.g. I believe the shorting of the AOI with 20km, we could do an erosion operation of 20km.
> > This would also impact the width of the AOI, but on the other hand i do not believe that would be a cause for problems in the processing.
> 
> This would likely be a more reliable approach than what I use currently. Do you have any ideas for how to erode the polygon?

This can be done with an affine transformation, no?

**Pull request merged**